### ISSUE TITLE
Not able to open dApp - Connect fails

### ISSUE THREAD

**Issue by christianbrb** [raiden-network/light-client#382]
## Steps to Reproduce
1. Navigate to https://lightclient.raiden.network/
2. Click on connect

## Expected Result
- I see the token screen and can use it

## Actual Result
- The token screen is briefly shown (0.5 seconds)

    ![Raiden-dApp-Google-Chrome-2019-10-11-09-11-59](https://user-images.githubusercontent.com/35398162/66632458-d17c7980-ec08-11e9-80e5-f15ec75f1a96.gif)

- Afterward, the start screen is shown
- Sometimes the following error is shown
    ![grafik](https://user-images.githubusercontent.com/35398162/66631881-7e55f700-ec07-11e9-9137-d39ee2838c0c.png)

- Error log

```
Failed to load resource: https://transport04.raiden.network/_matrix/client/r0/join/%23raiden_kovan_discovery%3Atransport04.raiden.network?access_token=MDAyOGxvY2F0aW9uIHRyYW5zcG9ydDA0LnJhaWRlbi5uZXR3b3JrCjAwMTNpZGVudGlmaWVyIGtleQowMDEwY2lkIGdlbiA9IDEKMDA1OWNpZCB1c2VyX2lkID0gQDB4Yjg0YzZiZmJhMTEwODU0M2VhNDdlOGI5YTY2OWNiMzlmZmM3MGY2YTp0cmFuc3BvcnQwNC5yYWlkZW4ubmV0d29yawowMDE2Y2lkIHR5cGUgPSBhY2Nlc3MKMDAyMWNpZCBub25jZSA9IERpSl5jLj1MVUNCWmFhZFYKMDAyZnNpZ25hdHVyZSBkWF8-jqQ8ym4HZY-B-jhr_mSSWnux2BG2FG1icXDH4wo the server responded with a status of 404 ()
```

**Comment by christianbrb** [comment]
After switching to Kovan

1. Click on Connect
2. I see the Token Screen
3. Signing two Matrix messages for transport03
4. Afterward, it goes back to the connect screen
5. Clicking again on Connect
6. I see the token screen
7. Signing two messages for transport04
8. Afterward it goes back to the connect screen
9. I see the token screen
10. Signing two Matrix messages for transport04
11. Afteward it goes back to the connect screen

Maybe related to #381

**Comment by christianbrb** [comment]
Now it worked - keeping it open to add debug information

**Comment by christianbrb** [comment]
Added the error code to the description
`
Failed to load resource: https://transport04.raiden.network/_matrix/client/r0/join/%23raiden_kovan_discovery%3Atransport04.raiden.network?access_token=MDAyOGxvY2F0aW9uIHRyYW5zcG9ydDA0LnJhaWRlbi5uZXR3b3JrCjAwMTNpZGVudGlmaWVyIGtleQowMDEwY2lkIGdlbiA9IDEKMDA1OWNpZCB1c2VyX2lkID0gQDB4Yjg0YzZiZmJhMTEwODU0M2VhNDdlOGI5YTY2OWNiMzlmZmM3MGY2YTp0cmFuc3BvcnQwNC5yYWlkZW4ubmV0d29yawowMDE2Y2lkIHR5cGUgPSBhY2Nlc3MKMDAyMWNpZCBub25jZSA9IERpSl5jLj1MVUNCWmFhZFYKMDAyZnNpZ25hdHVyZSBkWF8-jqQ8ym4HZY-B-jhr_mSSWnux2BG2FG1icXDH4wo the server responded with a status of 404 ()`

We need to work on #381

**Comment by andrevmatos** [comment]
Hmm, this is something very specific, not likely to happen on "normal operation" and not fixable by #381.
The issue is that transport04 is a fresh server, just recently fired. It happens that, while coding our transport, I expected the global rooms (in this case, `discovery`) would already have been created automatically by full clients connecting on this server, on this network. Well, transport04 being so fresh, it seems no one had logged with a full client on kovan on it at the moment this issue happened.
- The workaround is to wait for someone to do this, or do it one itself before logging in the light client.
- The quick fix is to catch the error when trying to join a non-existing global room, and let it be. discovery room is just used to force other transport servers to know us, and that'll happen anyway when we invite/are invited or interact with their users anyway. pfs global room is used only for some capacity updates which may be done by the hub eventually anyway as well.
- The proper fix is to follow the [join_global_room](https://github.com/raiden-network/raiden/blob/4cf077e06cc505e3718540a28289e2f56db3b4f0/raiden/network/transport/matrix/utils.py#L324-L323) raiden algorithm and create/alias the room ourselves if joining fails. Not very hard, just more code that wouldn't be used usually.

**Comment by christianbrb** [comment]
We should accommodate for a failure of the communication layer and go with:

> The proper fix is to follow the join_global_room raiden algorithm and create/alias the room ourselves if joining fails. Not very hard, just more code that wouldn't be used usually.

**Comment by christianbrb** [comment]
Seems to be fixed by #383
